ocaml/xapi/context.ml

```diff
@@ -309,11 +309,23 @@ let make ?(http_other_config = []) ?(quiet = false) ?subtask_of ?session_id
             " by task " ^ make_dbg [] "" subtask_of
         )
   ) ;
+  let span_details_from_task_name task_name =
+    match String.split_on_char ':' task_name with
+    | [x; y; uuid] when x = "dispatch" && y = "system.isAlive" ->
+        let span_name = x ^ ":" ^ y in
```

Comment:
This can avoid the repetition, it's not very readable, however:
```suggestion
    | ["dispatch"; "system.isAlive"; uuid] ->
        let span_name = String.sub task_name 0 23 in
```

Reply:
Yeah it is a little unreadable, I think I'll stick with the `when`